src/compiler/scala/tools/nsc/backend/jvm/opt/LocalOpt.scala

```diff
@@ -767,7 +770,7 @@ object LocalOptImpls {
       val index = local.index
       // parameters and `this` (the lowest indices, starting at 0) are never removed or renumbered
       if (index >= firstLocalIndex) {
-        if (!variableIsUsed(local.start, local.end, index)) localsIter.remove()
+        if (!used.contains(index)) localsIter.remove()
```

Comment:
I think this doesn't look right. A local variable slot in bytecode can be used for multiple local variables in source, e.g., in
```
if (foo) { val x = bar; x } else { val y = 1; 0 }
```
`x` and `y` may use the same local variable slot. In the local variable table, the entry for `y` is stale once the unused store is eliminated, even if the slot is in use by `x`. scalac doesn't reuse local variable slots, and the optimizer doesn't do any sort of register allocation either, but still this new implementation seems not ideal.

Isn't the underlying bug that the `(local.start, local.end)` rainge fails to include the `xSTORE` instruction you mention in the PR description?

Reply:
> Isn't the underlying bug that the (local.start, local.end) rainge fails to include the xSTORE instruction you mention in the PR description?

I've reworked this PR to search `(local.start.previousOrSelf, local.end)`. When I first wrote the PR I forgot that `InsnList` is doubly linked which led me to doing one extra full pass.